### PR TITLE
fix: updated PR#1077 - added sim check

### PR DESCRIPTION
**Pull request by alphaNewrex**
## Issue Fix
Fixes #1077 merging issue

## Description
Fixed the merging issue of PR # 1077

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by alphaNewrex** [comment]
@devansh-299 can you approve the workflow ?

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java [hunk 1]
```diff
@@ -75,8 +76,12 @@ public void onSimSelected(View view) {
     @OnClick(R.id.btn_confirm)
     public void onConfirmClicked() {
         dismiss();
-        if (getActivity() instanceof LinkBankAccountActivity) {
+        if (getActivity() instanceof LinkBankAccountActivity && (selectedSim ==1 || selectedSim == 2)){
             ((LinkBankAccountActivity) getActivity()).linkBankAccount(selectedSim);
         }
+        else
+        {
+            Toast.makeText(getContext().getApplicationContext(),getString(R.string.choose_a_sim),Toast.LENGTH_SHORT).show();
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java]: No need to use application context here. You can go through this [article](https://blog.mindorks.com/understanding-context-in-android-application-330913e32514) to learn more

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java [hunk 2]
```diff
@@ -75,8 +76,12 @@ public void onSimSelected(View view) {
     @OnClick(R.id.btn_confirm)
     public void onConfirmClicked() {
         dismiss();
-        if (getActivity() instanceof LinkBankAccountActivity) {
+        if (getActivity() instanceof LinkBankAccountActivity && (selectedSim ==1 || selectedSim == 2)){
```
Comment (devansh-299) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java]: Please insert a space between `)` & `{`. This is leading to a check-style violation. Also before making a pull request, do run the quality checking commands (_refer project README_) locally on your system.

**Review by devansh-299 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java (hunk 2 above)
Comment (alphaNewrex) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java]: Thanks I have updated according to the gradlew check report.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java (hunk 1 above)
Comment (alphaNewrex) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java]: I have updated this change and Thanks for the resource.

**Comment by alphaNewrex** [comment]
@devansh-299 please check the changes and approve the workflow. This was my first time using squash, so had to force push the squashed commit.

**Review by devansh-299 — APPROVED** [review]

**Pull request merged**